content/c-sharp/concepts/strings/terms/split/split.md

```diff
@@ -0,0 +1,136 @@
+
+## Syntax
+
+The various overloads of the `.Split()` method in C# are as follows:
+
+```pseudo
+Split(String[], Int32, StringSplitOptions)
+```
+
+Splits a string into a maximum number of substrings using the strings in an array as delimiters.
+
+```pseudo
+Split(Char[], Int32, StringSplitOptions)
+```
+
+Splits a string into a maximum number of substrings using the characters in an array as delimiters.
+
+```pseudo
+Split(String[], StringSplitOptions)
+```
+
+Splits a string into substrings based on the strings in an array.
+
+```pseudo
+Split(Char[])
+```
+
+Splits a string into substrings based on the characters in an array.
+
+```pseudo
+Split(Char[], StringSplitOptions)
+```
+
+Splits a string into substrings based on the characters in an array.
+
+```pseudo
+Split(Char[], Int32)
+```
+
+Splits a string into a maximum number of substrings based on the characters in an array. 
+
+- `StringSplitOptions`: Represents an enumeration that defines whether to include empty substrings `(StringSplitOptions.None)` or exclude them `(StringSplitOptions.RemoveEmptyEntries)`.
+- `int count`: Represents the maximum number of substrings to return.
+
+## Example
+
+The following examples demonstrates the use of `.Split()` method:
+
+```cs
+using System;
+
+class EqualsMethod {
+
+    public static void Main(string[] args)
+    {
+        string s1 = "Rivers, Mountains, Oceans";
+
+        string[] subs = s1.Split(',');
+        foreach (var sub in subs)
+        {
+          Console.WriteLine($"Substring: {sub}");
+        }
+
+       // To remove spaces, specify ', ' as the delimiter.
+        subs = s1.Split(", ");
+        foreach (var sub in subs)
+        {
+          Console.WriteLine($"Substring: {sub}");
+        }
+
+        // To limit the number of substrings to 2, specify the optional parameter
+        subs = s1.Split(", ", 2);
+        foreach (var sub in subs)
+        {
+          Console.WriteLine($"Substring: {sub}");
+        }
+    }
+}
+```
+
+This example results in the following output:
+
+```shell
+Substring: Rivers
+Substring:  Mountains
+Substring:  Oceans
+Substring: Rivers
+Substring: Mountains
+Substring: Oceans
+Substring: Rivers
+Substring: Mountains, Oceans
+```
+
+## Codebyte Example
+
+Run the following example to understand how the `.String()` method works:
+
+```codebyte/csharp
+using System;
+
+class EqualsMethod {
+
+    public static void Main(string[] args)
+    {
+        string s1 = "Rivers, Mountains, Oceans";
+
+        string[] subs = s1.Split(',');
+        foreach (var sub in subs)
+        {
+          Console.WriteLine($"Substring: {sub}");
+        }
+
+        // To remove spaces too, we can specify ', ' as the delimeter
+        subs = s1.Split(", ");
+        foreach (var sub in subs)
+        {
+          Console.WriteLine($"Substring: {sub}");
+        }
+    }
+}
+```
```

Comment:
This example is very similar to the example you have used above to explain the usage. Codebyte examples have to be a little different. You can try something like this - 

```cs
using System;

class SplitExample
{
    public static void Main(string[] args)
    {
        // A sample CSV string
        string csv = "John,25,Developer,New York,,john@example.com";

        // Split by comma (default behavior)
        string[] fields = csv.Split(',');
        Console.WriteLine("Fields after splitting by comma:");
        foreach (var field in fields)
            Console.WriteLine($"'{field}'");

        // Split by comma and remove empty entries
        string[] fieldsWithoutEmpty = csv.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
        Console.WriteLine("\nFields after removing empty entries:");
        foreach (var field in fieldsWithoutEmpty)
            Console.WriteLine($"'{field}'");

        // Split by multiple delimiters (comma and semicolon)
        string multiDelimiterString = "apple, orange; banana,grape";
        string[] fruits = multiDelimiterString.Split(new char[] { ',', ';' });
        Console.WriteLine("\nFruits after splitting by comma and semicolon:");
        foreach (var fruit in fruits)
            Console.WriteLine($"'{fruit.Trim()}'");  // Trim spaces
    }
}
```